packages/contracts/test/router/v6/nftx/listings.test.ts

```diff
@@ -0,0 +1,295 @@
+  let bob: SignerWithAddress;
+  let carol: SignerWithAddress;
+  let david: SignerWithAddress;
+  let emilio: SignerWithAddress;
+
+  let erc721: Contract;
+  let router: Contract;
+  let nftxModule: Contract;
+
+  beforeEach(async () => {
+    [deployer, alice, bob, carol, david, emilio] = await ethers.getSigners();
+
+    ({ erc721 } = await setupNFTs(deployer));
+
+    router = (await ethers
+      .getContractFactory("ReservoirV6_0_0", deployer)
+      .then((factory) => factory.deploy())) as any;
+      nftxModule = (await ethers
+      .getContractFactory("NFTXModule", deployer)
+      .then((factory) =>
+        factory.deploy(router.address, router.address)
+      )) as any;
+  });
+
+  const getBalances = async (token: string) => {
+    if (token === Sdk.Common.Addresses.Eth[chainId]) {
+      return {
+        alice: await ethers.provider.getBalance(alice.address),
+        bob: await ethers.provider.getBalance(bob.address),
+        carol: await ethers.provider.getBalance(carol.address),
+        david: await ethers.provider.getBalance(david.address),
+        emilio: await ethers.provider.getBalance(emilio.address),
+        router: await ethers.provider.getBalance(router.address),
+        nftxModule: await ethers.provider.getBalance(
+          nftxModule.address
+        ),
+      };
+    } else {
+      const contract = new Sdk.Common.Helpers.Erc20(ethers.provider, token);
+      return {
+        alice: await contract.getBalance(alice.address),
+        bob: await contract.getBalance(bob.address),
+        carol: await contract.getBalance(carol.address),
+        david: await contract.getBalance(david.address),
+        emilio: await contract.getBalance(emilio.address),
+        router: await contract.getBalance(router.address),
+        nftxModule: await contract.getBalance(nftxModule.address),
+      };
+    }
+  };
+
+  afterEach(reset);
+
+  const testAcceptListings = async (
+    // Whether to include fees on top
+    chargeFees: boolean,
+    // Whether to revert or not in case of any failures
+    revertIfIncomplete: boolean,
+    // Whether to cancel some orders in order to trigger partial filling
+    partial: boolean,
+    // Number of listings to fill
+    listingsCount: number
+  ) => {
+    // Setup
+
+    // Makers: Alice and Bob
+    // Taker: Carol
+    // Fee recipient: Emilio
+
+    let listings: SudoswapListing[] = [];
+    const feesOnTop: BigNumber[] = [];
+    for (let i = 0; i < listingsCount; i++) {
+      listings.push({
+        seller: getRandomBoolean() ? alice : bob,
+        nft: {
+          contract: erc721,
+          id: getRandomInteger(1, 100000),
+        },
+        price: parseEther(getRandomFloat(0.0001, 2).toFixed(6)),
+        isCancelled: false
+      });
+
+      if (chargeFees) {
+        feesOnTop.push(parseEther(getRandomFloat(0.0001, 0.1).toFixed(6)));
+      }
+    }
+
+    await setupNFTXListings(listings);
+
+    listings = listings.filter(_ => _.order);
```

Comment:
Why this filter? Can any orders be missing?

Reply:
for `isCancelled`

Reply:
Right, but we should have tests for how we handle cancelled listings (eg. if partial=false then we revert the whole transaction, if partial=true then filling would simply ignore cancelled listings).